bayesian/api/api_v2.py

```diff
@@ -202,6 +203,72 @@ def get(ecosystem, package, version):
 
         raise HTTPError(404, msg)
 
+    @staticmethod
+    def post():
+        """Handle the POST REST API call."""
+        input_json = request.get_json()
+        if not input_json:
+            raise HTTPError(400, error="Expected JSON request")
+        if not isinstance(input_json, dict):
+            raise HTTPError(400, error="Expected list of dependencies in JSON request")
+
+        ecosystem = input_json.get('ecosystem')
+        if not check_for_accepted_ecosystem(ecosystem):
+            msg = f"Ecosystem {ecosystem} is not supported for this request"
+            raise HTTPError(400, msg)
+
+        packages_list = []
+        response_template = namedtuple("response_template", ["message", "status"])
+
+        for pkg_obj in input_json.get('packageversions'):
+            package = pkg_obj.get('package')
+            version = pkg_obj.get('version')
+
+            if not all([ecosystem, package, version]):
+                raise HTTPError(422, "provide the valid input.")
+
+            if not validate_version(version):
+                return response_template(
+                    {'message': "Package version should not have special characters."}, 400)
+
+            if ecosystem == 'maven':
+                package = MavenCoordinates.normalize_str(package)
+
+            package = case_sensitivity_transform(ecosystem, package)
+
+            packages_list.append({"name": package, "version": version})
+
+        # Perform Component Analyses on Vendor specific Graph Edge.
+        analyses_result, unknown_pkgs = CABatchCall(
```

Comment:
@prashbnair i dont think this can be merged with ComponentAnalyses class, Reason:
- Signature of `CABatchCall` and `ComponentAnalyses` is different. CABatchCall accepts `packages:list` and `ecosystem:str` in its constructor whereas `ComponentAnalyses` accepts `eco:str`, `pkg:str` and `version:str` in its constructor.

- No Common methods that are used in each flow

Reply:
We already have a ComponentAnalysis class, we can have 2 methods that handle the single call and the batch call